Dynatrace/dynatrace-operator
# Remove proxy deprecation filter on OneAgent for argument and env var

## Description

[K8S-11311](https://dt-rnd.atlassian.net/browse/K8S-11311)

When removing a proxy from our DynaKube, we need to add `--set-proxy=""` argument to OneAgent pods to trigger its proxy configuration deletion. This PR removes a logic that only adds this arg and env var on specific OneAgent versions and set these always.

## How can this be tested?

- Deploy DynaKube with proxy
- Check that proxy is configured in OA pod (run on pod shell `cat /mnt/volume_storage_mount/var_log/os/ruxitagent_* | grep Proxy`)
- Remove proxy from DynaKube
- Check that `--set-proxy=""` is set in OA manifest
- Check that proxy configuration is gone (run on pod shell `cat /mnt/volume_storage_mount/var_log/os/ruxitagent_* | grep Proxy`)

NOTE: While testing it was noticed that proxy config may sometimes not be getting properly set in pod and not shown in the shell command even when should be configured (maybe based on tenants or different setups?)
